compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/peel_for_loop.mlir

```diff
@@ -46,17 +80,48 @@ func.func @no_peeling_example() -> i32 {
   return %r : i32
 }
 
-//      CHECK: #[[MAP:.*]] = affine_map<(d0, d1)[s0] -> (s0, d0 - d1)>
-//      CHECK: func @no_peeling_example(
-//      CHECK:   %[[C0_I32:.*]] = arith.constant 0 : i32
-//      CHECK:   %[[C0:.*]] = arith.constant 0 : index
-//      CHECK:   %[[C4:.*]] = arith.constant 4 : index
-//      CHECK:   %[[C4_0:.*]] = arith.constant 4 : index
-//      CHECK:   %[[RESULT:.*]] = scf.for %[[IV:.*]] = %[[C0]] to %[[C4_0]]
-// CHECK-SAME:       step %[[C4]] iter_args(%[[ACC:.*]] = %[[C0_I32]]) -> (i32) {
-//      CHECK:     %[[MIN:.*]] = affine.min #[[MAP]](%[[C4_0]], %[[IV]])[%[[C4]]]
-//      CHECK:     %[[CAST:.*]] = arith.index_cast %[[MIN]] : index to i32
-//      CHECK:     %[[ADD:.*]] = arith.addi %[[ACC]], %[[CAST]] : i32
-//      CHECK:     scf.yield %[[ADD]]
-//      CHECK:   }
-//      CHECK:   return %[[RESULT]]
+//      PEEL-FIRST: #[[MAP:.*]] = affine_map<(d0, d1)[s0] -> (s0, d0 - d1)>
+//      PEEL-FIRST: func @no_peeling_example(
+//      PEEL-FIRST:   %[[C0_I32:.*]] = arith.constant 0 : i32
+//      PEEL-FIRST:   %[[C0:.*]] = arith.constant 0 : index
+//      PEEL-FIRST:   %[[C4:.*]] = arith.constant 4 : index
+//      PEEL-FIRST:   %[[C4_0:.*]] = arith.constant 4 : index
+//      PEEL-FIRST:   %[[RESULT:.*]] = scf.for %[[IV:.*]] = %[[C0]] to %[[C4_0]]
+// PEEL-FIRST-SAME:       step %[[C4]] iter_args(%[[ACC:.*]] = %[[C0_I32]]) -> (i32) {
+//      PEEL-FIRST:     %[[MIN:.*]] = affine.min #[[MAP]](%[[C4_0]], %[[IV]])[%[[C4]]]
+//      PEEL-FIRST:     %[[CAST:.*]] = arith.index_cast %[[MIN]] : index to i32
+//      PEEL-FIRST:     %[[ADD:.*]] = arith.addi %[[ACC]], %[[CAST]] : i32
+//      PEEL-FIRST:     scf.yield %[[ADD]]
+//      PEEL-FIRST:   }
+//      PEEL-FIRST:   return %[[RESULT]]
+
+// -----
+
+#map = affine_map<(d0, d1)[s0] -> (s0, d0 - d1)>
+func.func @two_iteration_example() -> i32 {
+  %c0_i32 = arith.constant 0 : i32
+  %lb = arith.constant 0 : index
```

Comment:
Maybe to have better coverage, make a test (or modify this one) which doesn't start at a value for which lb % step != 0 ?

Reply:
Done.